pkg/util/pod/pod.go

```diff
@@ -56,3 +62,32 @@ func gateIndex(p *corev1.Pod, gateName string) int {
 		return g.Name == gateName
 	})
 }
+
+func ReadUIntFromLabel(obj client.Object, labelKey string) (*int, error) {
+	return ReadUIntFromLabelWithMax(obj, labelKey, math.MaxInt)
+}
+
+func ReadUIntFromLabelWithMax(obj client.Object, labelKey string, max int) (*int, error) {
+	value, found := obj.GetLabels()[labelKey]
+	kind := obj.GetObjectKind().GroupVersionKind().Kind
+	if !found {
+		return nil, fmt.Errorf("no label %q for %s %q", labelKey, kind, klog.KObj(obj))
+	}
+	intValue, err := readUIntFromStringWithMax(value, max)
+	if err != nil {
+		return nil, fmt.Errorf("incorrect label value %q for %s %q: %w", value, kind, klog.KObj(obj), err)
+	}
+	return intValue, nil
+}
+
+func readUIntFromStringWithMax(value string, max int) (*int, error) {
+	uintValue, err := strconv.ParseUint(value, 10, 0)
+	if err != nil {
+		return nil, err
```

Comment:
Hm, maybe we could re-package the error into `errInvalidInt` so that it is easier to assert?

IIUC this could work: `return nil, fmt.Sprintf("%w, message: %v", errInvalidInt, err.Error())`, then we could assert on  `errInvalidInt` rather than strings.

Reply:
cc @PBundyra as it may impact also your approach to testing

Reply:
Done

Reply:
Can you check if `return nil, fmt.Sprintf("%w, message: %v", errInvalidInt, err.Error())` works? I think it might be useful to get contextual information.